pkg/apis/kops/validation/legacy.go

```diff
@@ -512,6 +516,14 @@ func ValidateCluster(c *kops.Cluster, strict bool) field.ErrorList {
 
 	allErrs = append(allErrs, newValidateCluster(c)...)
 
+	if kops.CloudProviderID(c.Spec.CloudProvider) != kops.CloudProviderAWS {
```

Comment:
Wouldn't this otherwise work on other cloud providers when using VAULT_TOKEN?

Reply:
If you pass a static token, then I suppose this will work. Injecting a static token into each server in a safe way would be difficult though.

Reply:
I don't think it's a requirement to allow such customization, I just didn't want to foreclose it without consideration. It looks like it might be just as easy to extend kops for a new cloud provider as customize it with scripts.